### ISSUE TITLE
feature: ed25519 support

### ISSUE THREAD

**Issue by mattpallissard** [mirleft/ocaml-x509#137]
I receive the following error from ca certs in the global trust store.
```
Fatal error: exception (Failure "Unknown algorithm 1.3.101.112")
```
This error is encountered in downstream ocaml projects whether or not we are actually interacting with the ed25519 based / signed certs.

Full support would be great, but politely ignoring the certs would be a solid work around.  I'd imagine that there would also have to be work done in mirage-crypto.  Let me know if there is anything I can do to help, thanks in adavance.

**Comment by hannesm** [comment]
please be slightly more specific, there's preliminary support for encoding and decoding ED25519 private and public keys in https://github.com/mirleft/ocaml-x509/pull/136 -- would you need certificate support as well?

**Comment by hannesm** [comment]
Would you mind to specify which operating system / Linux distribution and version are you using?

**Comment by mattpallissard** [comment]
I built the package from your fork's master branch and it resolves the issue I was seeing, specifically with [opam-compiler](https://github.com/ocaml-opam/opam-compiler/issues/9).  Thank you.

For posterity sake
```
linux: arch
opam: 2.0.7
ocaml: The OCaml toplevel, version 4.11.1
opam-compiler: 0.1.1
```

**Comment by hannesm** [comment]
Hey @mattpallissard, could you use https://github.com/mirage/ca-certs/pull/11 and check whether this works (without the pinned `x509` branch)?

An `opam pin add ca-certs git+https://github.com/hannesm/ca-certs.git#resilient-trust-anchor-failures` (and `opam pin remove x509` to remove the x509 pin) should do that. Thanks.

**Comment by hannesm** [comment]
I merged and released ca-certs 0.1.3 as a stop-gap solution. This issue can stay open until we support ED25519 certificates (the referenced branch & PR contains some `assert false` and needs some work before being merged and released).

**Comment by mattpallissard** [comment]
Sorry for the delayed response and thanks.
It seems to work just fine.  If I pull the main branch and run the tests they pass as well, previously with v.0.1.2 the tests would fail.

**Comment by hannesm** [comment]
fixed in #145, thanks for the report. release coming up soon.
